docs/developer/design/authentication.rst

```diff
@@ -0,0 +1,154 @@
+Authentication
+==============
+
+Todo
+----
+
+- Provide uml diagrams for Dex + MetalK8s authentication flow
+
+Open questions
+--------------
+
+- Is support for Dex monitoring and alerting on Grafana
+  something we should think about now?
+
+Abstract
+--------
+
+To be able to identify who a user is and to manage(Create/Edit/Delete)
+these user identities, MetalK8s needs a middleman that acts in this light
+as an Identity Provider.
+
+To accomplish this, we introduce the concept of an OpenID Connect identity
+provider which has as role identity federation in a typical MetalK8s cluster.
+
+
+Scope
+-----
+
+The scope of **Authentication** will be split into 3 main iterations as
+detailed in the implementation section of this document.
+
+Requirements
+------------
+
+User Requirements for Authentication in MetalK8s has been discussed and is
+well documented. For more information, refer to -> [link to Requirements DOC]
+
+
+Design
+------
+
+For authentication in **MetalK8s**, we are going to use Dex, Dex
+connectors/plugins and it's integrated support for other
+External IDP Integration as an Identity Provider.
+
+Design choices
+^^^^^^^^^^^^^^
+
+Dex is chosen as an Identity Provider in MetalK8s based on the above
+requirements enforced by the following reasons:
+
+* Dex's support for multiple plugins greatly increases the potential for
+  integrating with existing user management system such as Active Directory,
+  LDAP, SAML and others.
+* The ability for Dex to issue tokens, revoke tokens, refresh tokens and even
+  control the lifetime of these tokens makes **Dex** suitable since we can
+  force users to re-authenticate when need arises.
+* Dex can be seamlessly deployed in a Kubernetes cluster.
+* Dex provides access to a highly customizable UI which is a step closer to
+  good user experience which we advocate for.
+* Dex provides support for multiple backends for storage. This guarantees user
+  account information persistence after upgrades.
```

Comment:
Not sure how that's useful. We use `etcd`, Dex can use `etcd` and is well integrated in Kubernetes (uses CRDs and such), as you mentioned 2 lines above, so I think the point is made.